src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -334,10 +409,7 @@ private void AddBindings(ArrayBuilder<BoundStatement> sectionBuilder, ImmutableA
                     {
                         var source = kv.Key;
                         var dest = kv.Value;
-                        var rewriter = this.LocalRewriter;
-                        sectionBuilder.Add(_factory.ExpressionStatement(
-                            rewriter.MakeStaticAssignmentOperator(
-                                _factory.Syntax, rewriter.VisitExpression(dest), rewriter.VisitExpression(source), RefKind.None, dest.Type, false)));
+                        sectionBuilder.Add(_factory.Assignment(dest, source));
```

Comment:
I do not see any non-default instances of KeyValuePair<BoundExpression, BoundExpression> created in this file. I see them created in DecisionTreeBuilder, but it doesn't look like DecisionTreeBuilder creates the nodes, it looks like it reuses nodes from the initial bound tree.